connectors/src/connectors/slack/chat/citations.ts

```diff
@@ -30,17 +28,14 @@ export function annotateCitations(
   for (const action of actions) {
     if (action && isRetrievalActionType(action) && action.documents) {
       action.documents.forEach((d) => {
-        references[d.reference] = {
-          reference: d.reference,
-          link: d.sourceUrl
-            ? d.sourceUrl
-            : makeDustAppUrl(
-                `/w/${d.dataSourceWorkspaceId}/builder/data-sources/${
-                  d.dataSourceId
-                }/upsert?documentId=${encodeURIComponent(d.documentId)}`
-              ),
-          title: getTitleFromRetrievedDocument(d),
-        };
+        // If the document has no sourceUrl we skip the citation.
+        if (d.sourceUrl) {
```

Comment:
I'll actually refactor this a bit more in a follow up PR, since we always have it now.

Reply:
Indeed this is merely a local stitch